### PR TITLE
(MODULES-2286) Create Tests for Basic DSC Resource "File"

### PR DESCRIPTION
**Pull request by cowofevil**
Add tests for "File" DSC resource type.

### CODE REVIEW

File: tests/acceptance/tests/basic_dsc_resources/file/file_valid_unicode_path.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+require 'erb'
+require 'dsc_utils'
+test_name 'MODULES-2286 - C68559 - Apply DSC File Resource with Valid Unicode "DestinationPath" Specified'
```
Comment (zreichert) [inline tests/acceptance/tests/basic_dsc_resources/file/file_valid_unicode_path.rb]: Duplicate test ID

**Comment by Iristyle** [comment]
Note that some of the negative tests that match Puppet error output are going to need to be modified as a result of MODULES-2292 and the new error output.

See https://github.com/puppetlabs/puppetlabs-dsc/commit/217b994e00325c95a29c626a399a830608a99093  for a couple of changes that were made to the suite to address the error message changes.

File: tests/acceptance/tests/basic_dsc_resources/file/negative/file_invalid_path.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  :dsc_ensure          => 'Present',
+  :dsc_destinationpath => 'K:\test.file',
+  :dsc_contents        => 'You will never see this!',
+}
+
+dsc_manifest_template_path = File.join(local_files_root_path, 'basic_dsc_resources', 'dsc_single_resource.pp.erb')
+dsc_manifest = ERB.new(File.read(dsc_manifest_template_path), 0, '>').result(binding)
+
+# Verify
+error_msg = /returned 1: Invoke-CimMethod : No such file or directory/
```
Comment (Iristyle) [inline tests/acceptance/tests/basic_dsc_resources/file/negative/file_invalid_path.rb]: The error message from Puppet is now

```
Error: No such file or directoryThe related file/directory is: K:\test.file.  The related ResourceID is [File]DirectResourceAccess.
Error: /Stage[main]/Main/Dsc_file[file_test]/ensure: change from absent to present failed: No such file or directoryThe related file/directory is: K:\test.file.  The related ResourceID is [File]DirectResourceAccess.
```

File: tests/acceptance/tests/basic_dsc_resources/file/negative/file_invalid_sourcepath.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  :dsc_ensure          => 'Present',
+  :dsc_destinationpath => 'C:\test.file',
+  :dsc_sourcepath      => 'K:\source\does_not.exist',
+}
+
+dsc_manifest_template_path = File.join(local_files_root_path, 'basic_dsc_resources', 'dsc_single_resource.pp.erb')
+dsc_manifest = ERB.new(File.read(dsc_manifest_template_path), 0, '>').result(binding)
+
+# Verify
+error_msg = /returned 1: Invoke-CimMethod : SourcePath must be accessible for current configuration/
```
Comment (Iristyle) [inline tests/acceptance/tests/basic_dsc_resources/file/negative/file_invalid_sourcepath.rb]: The error message from Puppet is now

```
Error: SourcePath must be accessible for current configuration. The related file/directory is: K:\source\does_not.exist.  The related ResourceID is [File]DirectResourceAccess.
Error: /Stage[main]/Main/Dsc_file[file_test]/ensure: change from absent to present failed: SourcePath must be accessible for current configuration. The related file/directory is: K:\source\does_not.exist.  The related ResourceID is [File]DirectResourceAccess.
```

**Comment by cowofevil** [comment]
@Iristyle @zreichert Fixed, ready for review again.

**Comment by zreichert** [comment]
LGTM +1

**Pull request merged**